### DISCUSSION TITLE
Povratna informacija na prijavu projekta (Tim PI2247)

### DISCUSSION THREAD

**Discussion by zstapic** [foivz/pi22-igadzic-dbrajdic-dmujanovi#1] [Announcements]
Poštovani kolege,

oznaka vašeg tima je PI2247, a vaš mentor sam ja.

Tema vašeg prijavljenog projekta predstavlja sustav koji više skoro i nema primjenu u praksi, pa vas molim da vašu prijavu dopunite kako bi videoteka postala e-videoteka, to jest platforma za streamanje filmova bilo kojeg korisnika bilo kojem drugom korisniku. Osnovni UC bi bio da bilo koji korisnik u vašu platformu može prijaviti film koji ima (film bi bio pohranjen na računalu vlasnika), te bi drugi korisnici mogli posuditi, to jest zahtijevati stream filma, ali samo dok je računalo i odgovarajući windows servis kod vlasnika filma upaljen. To bi osiguralo vrlo dinamičnu okolinu gdje bi krajnji korisnici praktički kod svakog dolaska u aplikaciju vidjeli potencijalno drugačije dostupne filmove samo kod onih korisnika koji trenutno imaju uključenu opciju pružanja streaminga. Tehnološki jedini izazov je samo streamanje filmova preko desktop aplikacije, ali već postoje rješenja kao npr. ovo (http://csharpmagics.blogspot.com/2008/11/stream-video-over-lan-using-cnet.html). 

Većina vaših prijavljenih funkcionalnosti vrijedi i za ovaj sustav, a promjene su potrebne samo na nekim funkcionalnostima (npr. F04), neke funkcionalnosti je potrebno dodati (npr. windows servis za pružanje filma...), a neke možete obrisati (npr. F07, F08 i F10). Dakle, vlasnik filma bi trebao film upisati u aplikaciju / bazu, te moći imati uvid u trenutno akitvan broj gledatelja svakog svog filma, kao i moći pokrenuti/zaustavit servis. 

Za potrebe kolegija, dovoljno je da projekt radi u lokalnom LAN-u, jer inače bi bilo potrebno imati i online server, što u ovom slučaju možemo izbjeći.

Baza podataka na vašem projektu obvezno mora biti centralizirana (na serveru), a pristup bazi ćemo vam omogućiti mi nastavnici. Molim odgovorite ovdje u diskusiju da ste primili poruku, a potom i nakon što napravite tražene dorade.

Rok za promjene prijave neka bude nedjelja, 3.4.2022.

**Comment by zstapic** [comment]
I na ovo sam naletio: https://github.com/temmyakins/Video-Streaming-using-AIMD-Algorithm-over-UDP

**Comment by dmujanovic** [comment]
Poštovani profesore, 
slažemo se sa Vama i da je ideja malo zastarjela, tako da ćemo se potruditi realizirati platformu e-videoteke. Trenutno radimo na promjenama funkcionalnosti kako bi bile adekvatne temi.

**Comment by zstapic** [comment]
Konzultacije, 13.4.2022.
Dogovoreno:
 - Ivan Gadžić - Dodati opciju odobravanja novih filmova
 - Dominik Mujanović - omogućiti neregistriranim korisnicima unos filmova
 - Dominik Brajdić - omogućiti gledanje samo registriranim korisnicima, a pregled svima. Streamanje filma preoblikovati u gledanje filma bez interakcije s vlasnikom filma. Upravljanje korisnicima složiti da se bilo koji korisnik može proglasiti administratorom i/ili recenzentom (za odobravanje filmova).

Lp, 
Zlatko S.
